### DISCUSSION TITLE
Scratch Programming with STELLA

### DISCUSSION THREAD

**Discussion by willabe-13** [STELLA-Landsat/STELLA#43] [Ideas]
Hi All, Am fairly new to this but wanted to inquire about using coding blocks like scratch for students to get better involved in the programming and logistics side of things. It might be that the two things are virtually incompatible but we might also be delightfully surprised. 

For instance students might be able to integrate calibration standards on a time basis and learn how to account for measurement differences due to barometric pressure or temperature or time of day. 

A lot of the libraries for Raspberry Pi and Arduino overlap with Scratch are built out and it would be more a consideration of specific sub-libraries for all things spectroscopy.

Maybe this exists and just havent come across it yet...

**Comment by paulmirel** [comment]
Hello! 

I haven't done any programming in Scratch, but I am aware of the platform. If Scratch can open files, then there are lots of possibilities for post processing the data taken by STELLA. The files are in .CSV format, on micro SD cards. I'll be interested to hear if you have any luck opening them in Scratch. 

Programming for the instrument is something we do in CircuitPython, and it tightly constrained in terms of RAM. I expect that attempts to alter or reprogram the instrument itself using Scratch will be unsuccessful. 

P
